packages/angular_devkit/build_angular/src/differential-loading/utils.ts

```diff
@@ -0,0 +1,64 @@
+/**
```

Comment:
To determine whether deferential loading is needed it should be simple;

```ts
// In browser builder we already know if the it supports `supportES2015` as we use this to populate the correct `mainFields`
export function isDifferentialLoadingNeeded(projectRoot: string, supportES2015: boolean): boolean {
  const browsersList: string[] = browserslist(undefined, {path: projectRoot});
  return supportES2015 && !caniuse.isSupported('es6-module', browsersList.join(', '));
}
```

or
```ts
// In browser builder we already know the target
export function isDifferentialLoadingNeeded(projectRoot: string, target: ts.ScriptTarget): boolean {
  const supportES2015 = target !== projectTs.ScriptTarget.ES3 && target !== projectTs.ScriptTarget.ES5;
  const browsersList: string[] = browserslist(undefined, {path: projectRoot});
  return supportES2015 && !caniuse.isSupported('es6-module', browsersList.join(', '));
}
```

Reply:
I see your point. Thank you!